CONTRIBUTIONS.adoc

```diff
@@ -0,0 +1,269 @@
+
+== Code contributions
+
+If you have fixed a bug or implemented an enhancement, you can
+contribute your changes via GitHub's pull requests. This is not
+restricted to code, on the contrary, fixes and enhancements to
+link:#documentation[documentation] and link:#tests[tests] alone are also
+very valuable.
+
+=== Choosing something to work on
+
+Often you already have a bug or an enhancement you want to work on in
+your mind, but you can also look at the link:[issue tracker] to find
+bugs and enhancements submitted by others. The issues vary significantly
+in complexity and difficulty, so you can try to find something that
+matches your skill level and knowledge. There are two specific labels to
+look for when looking for something to contribute:
+
+https://github.com/robotframework/RIDE/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+
+
+`good first issue`::
+  These issues typically do not require any knowledge of Robot Framework
+  internals and are generally easy to implement or fix. Thus these
+  issues are especially good for new contributors.
+
+  
+https://github.com/robotframework/robotframework/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22
+
+`help wanted`::
+  These issues require external help to get implemented or fixed.
+
+=== Pull requests
+
+On GitHub pull requests are the main mechanism to contribute code. They
+are easy to use both for the contributor and for the person accepting
+the contribution, and with more complex contributions it is easy also
+for others to join the discussion. Preconditions for creating pull
+requests are having a https://github.com/[GitHub account], installing
+https://git-scm.com[Git] and forking the link:[Robot Framework project].
+
+GitHub has good articles explaining how to
+https://help.github.com/articles/set-up-git/[set up Git],
+https://help.github.com/articles/fork-a-repo/[fork a repository] and
+https://help.github.com/articles/using-pull-requests[use pull requests]
+and we do not go through them in more detail. We do, however, recommend
+to create dedicated topic branches for pull requests instead of creating
+them based on the master branch. This is especially important if you
+plan to work on multiple pull requests at the same time.
+
+=== Coding conventions
+
+==== General guidelines
+
+Robot Framework uses the general Python code conventions defined in
+https://www.python.org/dev/peps/pep-0008/[PEP-8]. In addition to that,
+we try to write
+http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html[idiomatic
+Python] and follow the
+https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)[SOLID
+principles] with all new code. An important guideline is that the code
+should be clear enough that comments are generally not needed.
+
+All code, including test code, must be compatible with all supported
+Python interpreters and versions. Most importantly this means that the
+code must support both Python 2 and Python 3.
```

Comment:
>must support both Python 2 and Python 3.
must support Python 3 (actually, greater than 3.6). RIDE's code still have some Python 2 conditioning, due to its dual compatibility on version 1.7.4.2, and also in the included Robot Framework library from version 3.1.2.